### PR TITLE
feat: spoof-wifi-connection patch

### PR DESCRIPTION
**Pull request by Linus789**
Related patches PR: [revanced-patches#1527](https://github.com/revanced/revanced-patches/pull/1527)

### CODE REVIEW

File: app/src/main/java/app/revanced/all/connectivity/fakewifi/FakeWifiPatch.java [hunk 1]
```diff
@@ -0,0 +1,380 @@
+package app.revanced.all.connectivity.fakewifi;
```
Comment (oSumAtrIX) [inline app/src/main/java/app/revanced/all/connectivity/fakewifi/FakeWifiPatch.java:1]: This class should be in the package `app.revanced.all.connectivity.wifi.fake`. Make sure to apply this in the related patch as well.

**Review by oSumAtrIX — COMMENTED** [review]
Many code blocks contain a lot of duplicate code. This might be intentional to accommodate for readability and logical separation between the separate methods. If that is not the case, duplicate code should be consolidated.

**Comment by Linus789** [comment]
Took a stab at removing some duplicate logic.

File: app/src/main/java/app/revanced/all/connectivity/wifi/spoof/SpoofWifiPatch.java [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+import android.net.NetworkInfo;
+import android.net.NetworkRequest;
+import android.os.Build;
+import android.os.Handler;
+
+import androidx.annotation.RequiresApi;
+
+public class SpoofWifiPatch {
+
+    // Used in order to check what the (real or fake) active network is (take a look at `hasTransport`)
```
Comment (oSumAtrIX) [inline app/src/main/java/app/revanced/all/connectivity/wifi/spoof/SpoofWifiPatch.java:17]:
```suggestion
    // Used in order to check what the (real or fake) active network is (take a look at `hasTransport`).
```

**Review by oSumAtrIX — COMMENTED** [review]
There is still a fair chunk of duplicate code left, though reducing it would take a toll on readability. For that reason, I'd say this is ready to get merged if you agree with me.

**Comment by Linus789** [comment]
> There is still a fair chunk of duplicate code left, though reducing it would take a toll on readability. For that reason, I'd say this is ready to get merged if you agree with me.

I agree.

**Review by oSumAtrIX — DISMISSED** [review]

**Pull request merged**